src/vue/fields/DateField.vue

```diff
@@ -93,139 +119,226 @@ export default {
   },
 
   watch: {
-    'value': function () {
-      this.flatpickrDate = this.value
+    /**
+     * Watch for changes from parent component and update DOM
+     *
+     * @param {string|date} newValue
+     */
+    value (newValue) {
+      // Prevent updates if v-model value is same as input's current value
+      if (newValue === this.flatpickrDate) return
+      // Sets the current selected date after value changed
+      if (this.fp) this.fp.setDate(newValue, true)
+    },
+    /**
+     * Watch for any config changes and redraw date-picker
+     *
+     * @param {object} newConfig
+     */
+    config: {
+      deep: true,
+      handler (newConfig) {
+        let safeConfig = Object.assign({}, newConfig)
+        // Workaround: Don't pass hooks to configs again otherwise
+        // previously registered hooks will stop working
+        // Notice: we are looping through all events
+        // This also means that new callbacks cannot be passed once a component
+        // has been initialized
+        ALLOWED_FLATPICKR_EVENTS.forEach((hook) => {
+          delete safeConfig[hook]
+        })
+
+        // set new config
+        this.fp.set(safeConfig)
+      },
     },
   },
 
-  created () {
-    this.flatpickrDate = this.value
+  mounted () {
+    // Return early if flatpickr is already mounted
+    if (this.isFlatpickrMounted) return
+
+    let safeConfig = Object.assign({}, this.config)
+
+    // Inject defined methods into events array
+    Object.values(BUILDIN_EVENTS).forEach(hook => {
+      // eslint-disable-next-line
+      safeConfig[hook] = this.arrayify(safeConfig[hook] || [])
+        .concat((...args) => this[hook](...args))
+    })
```

Comment:
1. Is not `// eslint-disable-next-line` useless?
2. `this.arrayify(safeConfig[hook] || [])` — arrayify should handle `undefined` by itself. It is strange that arraify accepts array
3. the bunch of code could be extracted to somehting like "hookFlatpickrHooks" or "catchFlatpickrHooks"

Reply:
I think we do not need to create a new method for it. We have 3 lines of code with understandable docs that used only once.